DuckDuckGo/VPNWaitlistView.swift

```diff
@@ -0,0 +1,387 @@
+//
+//  VPNWaitlistView.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#if NETWORK_PROTECTION
+
+import SwiftUI
+import Core
+import Waitlist
+import DesignResourcesKit
+
+@available(iOS 15.0, *)
+struct VPNWaitlistView: View {
+
+    @EnvironmentObject var viewModel: WaitlistViewModel
+
+    var body: some View {
+        switch viewModel.viewState {
+        case .notJoinedQueue:
+            VPNWaitlistSignUpView(requestInFlight: false) { action in
+                Task { await viewModel.perform(action: action) }
+            }
+        case .joiningQueue:
+            VPNWaitlistSignUpView(requestInFlight: true) { action in
+                Task { await viewModel.perform(action: action) }
+            }
+        case .joinedQueue(let state):
+            VPNWaitlistJoinedWaitlistView(notificationState: state) { action in
+                Task { await viewModel.perform(action: action) }
+            }
+        case .invited:
+            VPNWaitlistInvitedView { action in
+                Task { await viewModel.perform(action: action) }
+            }
+        case .waitlistRemoved:
+            fatalError("State not supported for VPN waitlists")
+        }
```

Comment:
The `waitlistRemoved` state was implemented for Windows which shows a different view after the waitlist is over, but we don't support it since this flow will not be accessible at all after the waitlist test.